std/demangle.d

```diff
@@ -18,72 +18,73 @@
  */
 module std.demangle;
 
-/+
-private class MangleException : Exception
+/**
+Demangle D mangled names.
+
+Params:
+    name = the mangled name
+Returns:
+    A `string`. If it is not a D mangled name, it returns its argument name.
+ */
+string demangle(string name) @safe pure nothrow
 {
-    this()
-    {
-        super("MangleException");
-    }
+    import core.demangle : demangle;
+    auto ret = demangle(name);
+    return ret.idup;
```

Comment:
What's wrong with `assumeUnique`? ok, the name doesn't fit, but casting to string should be fine.

If you want to keep `idup`, I'd recommend not to make a copy if the name was not demangled (as the documentation "returns its argument" can be interpreted).